tests/migrations/test_state.py

```diff
@@ -920,6 +920,32 @@ class Meta:
         choices_field = Author._meta.get_field('choice')
         self.assertEqual(list(choices_field.choices), choices)
 
+    def test_choices_unroll_iterator(self):
```

Comment:
General support for having an iterator in `choices` isn't new. I'm don't think this test is necessary at all if we already test the behavior for iterables.